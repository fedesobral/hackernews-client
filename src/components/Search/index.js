import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Search extends Component {

  componentDidMount() {
    this.input.focus();
  }

  render() {
    const { value, onChange, onSubmit, children } = this.props;
    return (
      <form onSubmit={onSubmit}>
        <input  type='text'
                value={value}
                onChange={onChange}
                ref={(node) => { this.input = node; }} />
        <button type='submit'>
          {children}  
        </button>
      </form>
    );
  }
}

Search.propTypes = {
  value: PropTypes.string,
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  children: PropTypes.node.isRequired
}



// const Search = ({ value, onChange, onSubmit, children }) => {
//   return (
//     <form onSubmit={onSubmit}>
//       <input  type='text'
//               value={value}
//               onChange={onChange} />
//       <button type='submit'>
//         {children}  
//       </button>
//     </form>
//     );
// }
export default Search;