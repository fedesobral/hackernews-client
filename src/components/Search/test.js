import React from 'react';
import ReactDOM from 'react-dom';
// import { shallow, render, mount } from 'enzyme';
// import { renderToJson } from 'enzyme-to-json';
import Search from '../Search';

describe('Search', () => {

	it('renders', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<Search>Search</Search>, div);
	});

	// test('snapshots', () => {
	// 	// const createNodeMock = function () {
	// 	//   // You can return anything from this function.
	// 	//   // For example:
	// 	//   return focus() {
	// 	//       // Do nothing
	// 	//   }
	// 	// }
	// 	const component = render(
	// 		<Search onSubmit={jest.fn()}>Search</Search>
	// 	);
	// 	const tree = renderToJson(component);
	// 	expect(tree).toMatchSnapshot();
	// });

	// test('events', () => {
	// 	const props = { 
	// 		value: 'search',
	// 		onSubmit: jest.fn(),
	// 		onChange: jest.fn()
	// 	};
	// 	const element = mount(
	// 		<Search { ...props }>Search</Search>
	// 	);

 //    	expect(props.onChange).not.toBeCalled();
	// 	element.find('input[type="text"]').simulate('change');
	// 	expect(props.onChange).toHaveBeenCalledTimes(1);

 //    	expect(props.onSubmit).not.toBeCalled();
	// 	element.find('form').simulate('submit');
	// 	expect(props.onSubmit).toHaveBeenCalledTimes(1);
	// });
});
