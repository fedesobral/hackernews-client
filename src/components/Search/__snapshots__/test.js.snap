exports[`Search snapshots 1`] = `
<form>
  <input
    type="text" />
  <button
    type="submit">
    Search
  </button>
</form>
`;
